### PR TITLE
Client theme

### PR DESCRIPTION
**Pull request by GabrielInTheWorld**
- Added the @angular/material-package to project.
- Added themes to the project.
- Changed components to MD-components.

### CODE REVIEW

File: angular-tour-of-heroes/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -3,17 +3,36 @@ import { NgModule } from '@angular/core';
 import {RouterModule, Routes} from '@angular/router';
 
 // internal components
-import {HeroesComponent} from './heroes/heroes.component';
-import {DashboardComponent} from './dashboard/dashboard.component';
-import {HeroDetailComponent} from './hero-detail/hero-detail.component';
+import { routes as routeComponents } from '../environments/routes';
+// import {HeroesComponent} from './heroes/heroes.component';
```
Comment (jsaalfeld) [inline angular-tour-of-heroes/src/app/app-routing.module.ts]: you should remove unneccary comments

File: angular-tour-of-heroes/e2e/src/app.po.ts [hunk 2]
```diff
@@ -1,5 +1,8 @@
 import { browser, by, element } from 'protractor';
 
+/**
+ * @ignore
```
Comment (jsaalfeld) [inline angular-tour-of-heroes/e2e/src/app.po.ts]: not a too good style, to just ignore `compodoc`s requests ;)

File: angular-tour-of-heroes/angular.json [hunk 3]
```diff
@@ -131,5 +131,10 @@
...
     }
   },
-  "defaultProject": "angular-tour-of-heroes"
+  "defaultProject": "angular-tour-of-heroes",
+  "schematics": {
+    "@schematics/angular:component": {
+      "styleext": "scss"
+    }
+  }
 }
```
Comment (jsaalfeld) [inline angular-tour-of-heroes/angular.json]: you should add a newline at the end

File: angular-tour-of-heroes/src/app/hero-detail/hero-detail.component.html [hunk 4]
```diff
@@ -1,20 +1,16 @@
...
-  </div>
-  <!-- delete button -->
-  <div class="button" *ngIf="!isCreating">
-    <button class="button-error" (click)="deleteHero()">Delete</button>
+  <app-big-input [label]="label" [placeholder]="placeholder" [value]="hero.name" (changed)="onChange($event)"></app-big-input>
+  <div class="gm-hero-button-bar">
+    <!-- back button -->
+    <button mat-button (click)="goBack()">Go back</button>
+    <!-- submit button -->
+    <button mat-button color="primary" *ngIf="isCreating && hero.name !== ''" (click)="createNewHero()">Submit</button>
```
Comment (jsaalfeld) [inline angular-tour-of-heroes/src/app/hero-detail/hero-detail.component.html]: styles should be added via the `scss` file, not directly in html

File: angular-tour-of-heroes/src/app/hero-nav-bar/hero-nav-bar.component.html [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      <a *ngFor="let route of iRoutes" mat-list-item routerLink="{{ route.link }}">
+        <mat-icon>{{ route.icon }}</mat-icon>
+        <span>{{ route.title }}</span>
+      </a>
+    </mat-nav-list>
+  </mat-sidenav>
+  <mat-sidenav-content>
+    <mat-toolbar id="gmToolbar" class="mat-primary">
+      <button
+        type="button"
```
Comment (jsaalfeld) [inline angular-tour-of-heroes/src/app/hero-nav-bar/hero-nav-bar.component.html]: same here, you should add the styles via the `scss` file, not the `html`

File: angular-tour-of-heroes/src/app/hero-nav-bar/hero-nav-bar.component.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+})
+export class HeroNavBarComponent {
+
+  /**
+   * The routes for the navigation-drawer.
+   */
+  iRoutes: Route[] = routes;
+
+  /**
+   * Don't know, what this do.
```
Comment (jsaalfeld) [inline angular-tour-of-heroes/src/app/hero-nav-bar/hero-nav-bar.component.ts]: then you should learn about this

**Comment by GabrielInTheWorld** [comment]
Please try to compile the project again.
The second time this error-message isn't shown.

**Pull request merged**